lua/telescope/pickers/layout_strategies.lua

```diff
@@ -640,13 +640,32 @@ layout_strategies.vertical = make_documented_layout(
 
     local height_padding = math.floor((max_lines - height) / 2)
     if not layout_config.mirror then
-      preview.line = height_padding + bs + 1
-      results.line = (preview.height == 0) and preview.line or preview.line + preview.height + (1 + bs)
-      prompt.line = results.line + results.height + (1 + bs)
+      preview.line = height_padding + (1 + bs)
+      if layout_config.prompt_position == "top" then
+        prompt.line = (preview.height == 0) and preview.line or preview.line + preview.height + (1 + bs)
+        results.line = prompt.line + prompt.height + (1 + bs)
+      elseif layout_config.prompt_position == "bottom" then
+        results.line = (preview.height == 0) and preview.line or preview.line + preview.height + (1 + bs)
+        prompt.line = results.line + results.height + (1 + bs)
+      else
+        error(
+          "Unknown prompt_position: " .. tostring(self.window.prompt_position) .. "\n" .. vim.inspect(layout_config)
+        )
```

Comment:
I usually suggest `string.format`, it should make it a little bit more readable and you dont need `tostring`.

But isnt that big of a deal :laughing: